adobe/react-spectrum
# fix(#1961/#2958): Manage focus in docs popover breadcrumbs

When a link within a popover changes the contents of the popover, focus will be lost to the document.body, because the element that had focus will be removed from the DOM and replaced with new content.

There was prior work done on this, but we had to change some CSS to do it, we should figure out a way that doesn't require any CSS changes to the component. So this is really about managing the focus in our docs, not necessarily about Breadcrumbs themselves.
#728

We can make changes to FocusScope to better account for a content changes within the Focus resulting in a loss of focus while the scope remains active, as when the contents of a Dialog change due to actions taken within the Dialog, but for the Breadcrumbs use case, for a consistent user experience, it seems that the current BreadcrumbItem still needs to be focused explicitly.


Closes #1961 

Related issue/PR:  #2592/#2594 This related PR addresses the problem when the current breadcrumb changes from an action within the Breadcrumbs, but it can't account for explicitly focusing the current BreadcrumbItem after a content change initiated from elsewhere. I think we still have to handle that within our docs.

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issues #1961 and #2958.

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/99eb47497ee388f61e737fded0fcdfb8a0433bbe/docs/react-aria/useMenu.html
2. Tab to `AriaMenuOptions` link in the [API code example](https://reactspectrum.blob.core.windows.net/reactspectrum/99eb47497ee388f61e737fded0fcdfb8a0433bbe/docs/react-aria/useMenu.html#api).
3. Press Enter to expand `AriaMenuOptions` popover.
4. BreadcrumbItem for the current location, `AriaMenuOptions`, should have focus 
5. Drill down to AriaMenuOptions > CollectionChildren > CollectionElement > SectionElement > SectionProps > ItemElement > ItemProps. The dropdown menu should display. After clicking each link to refresh the contents of the popover, the  BreadcrumbItem for the current location should automatically focus.
6. Open dropdown and arrow down to focus, then press space to activate CollectionElement.
CollectionElement BreadcrumbItem element should autoFocus if things are working correctly, otherwise focus will be lost to the document.body, and the next tab stop will be and element outside of the popover.

## 🧢 Your Project:

Adobe/Accessibility
